src/Dialogs/Weather/NOAADetails.cpp

```diff
@@ -78,7 +78,7 @@ NOAADetailsWidget::Update()
 
   SetText(metar_taf.c_str());
 
-  StaticString<100> caption;
+  StaticString<140> caption;
```

Comment:
The compiler in LTO mode determined the exact length of all string variables which can go into the ...sprintf function.
Thus the maximum length of the string buffer is deterministic.

Reply:
The parameter strings cannot possibly be known at compile time.
We cannot avoid truncation. We can only make the buffer reasonably large that truncation does not affect the user, for example because the available screen size for the text isn't large enough anyway.
Your change is not based on whether this affects truncation; it seems your change is only motivated by a bogus compiler warning, which is the wrong kind of motivation.

Reply:
OK. Than rather truncate the parameter strings in order to fit into the buffer without risk of overflow?

Reply:
Why truncate parameter strings?
What "risk of overflow" do you mean?